Experiment text file controls push_config(event)

miraynamc opened the pull request:
Modified push_config(event) to take configurations from experiment_main.txt

miraynamc committed:
Experiment text file controls push_config(event)

Modified push_config(event) to take configurations from experiment_main.txt

anaymehta commented on dyno/panda/dyno_panel.py:
```diff
@@ -110,32 +110,35 @@ def map(x, in_min, in_max, out_min, out_max):
+          if pwm < 0:
             pwm = 0
-        elif pwm > 9:
+           elif pwm > 9:
```
Indentation looks a bit off to me here, everything else looks great

miraynamc replied:
Thank you for pointing that out! It's been fixed

miraynamc committed:
Edited push_config(event) to wait for new instructions

Push_config event will be expecting the instruction file to be edited/ updated with new instructions. Unless instructions say "end" the experiment, push_config(event) will wait for more

miraynamc committed:
Pseudocode for write_instruction

ashvinnagarajan commented on dyno/panda/dyno_panel.py:
```diff
@@ -110,32 +110,53 @@ def map(x, in_min, in_max, out_min, out_max):
             pwm = 9
         pwm = str(map(pwm, 0, 9, 0, 255)).zfill(3)
         magnet_config = magnet_config + str(i) + ":" + pwm + ","
-    magnet_config = magnet_config[:-1] + ";"
-    if READ_SERIAL_ON:
-        driver.write(magnet_config)
-    print(magnet_config)
-
-    throttle = int(throttle_slider.get())
-    if throttle < 0:
-        throttle = 0
-    elif throttle > 100:
-        throttle = 100
-    throttle = str(map(throttle, 0, 100, 0, 255)).zfill(3)
-    throttle_config = "motor=" + throttle + ";"
-    if READ_SERIAL_ON:
-        driver.write(throttle_config)
-    print(throttle_config)
-    mcu_acknowledged=False
```
Ahh no please don't delete this hahah this is good code

miraynamc replied:
Whoops! Got it 👍

ashvinnagarajan reviewed (changes requested):
Hey guys this is a great start! Instead of deleting the existing functionality, we are going to expand the code to also allow for "recipes". Docopt allows us to pass in files as a parameter so we are only going to run the recipe code IF such a file has been given. Otherwise, we will default to the real-time setup that is written currently.
